lib/experimental/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -173,6 +173,8 @@ public static function get_merged_data( $origin = 'custom' ) {
 		if ( 'custom' === $origin ) {
 			$result->merge( static::get_user_data() );
 		}
+		// Generate the default spacing sizes presets.
+		$result->set_spacing_sizes();
```

Comment:
If the goal is to create a new preset, why doesn't go through the normal data layers (both the preset and any option to enable/disable)?

- core can provide spacing presets
- theme can provide spacing presets
- users can provide its own as well

By not respecting the algorithm, things like making global styles data filterable becomes more difficult, see https://github.com/WordPress/gutenberg/pull/44015

@glendaviesnz @andrewserong @ramonjd @aaronrobertshaw

Reply:
That comment is a little misleading, it isn't actually generating the presets, the spacing presets are generated via the standard flow using [`PRESETS_METADATA`](https://github.com/WordPress/gutenberg/blob/trunk/lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php#L1145).

This method is actually generating the `spacingSizes` array, which is the equivalent of the  `fontSizes` array ... but unlike `fontSizes` the core default for spacing is to autogenerate the array from a set of [scale values](https://github.com/WordPress/gutenberg/blob/trunk/lib/compat/wordpress-6.1/theme.json#L334). 

The idea was that themes or users could override some or all of the scale values, eg. at some point in global styles there will be a UI where a user may just update the `increment` value - so generation of the `spacingSizes` needs to happen after the core/theme/user spacingScale values are merged - hence it's location here.

But, maybe we need to rethink this if this is going to hinder the filters you mention? I will update the comments to avoid confusion, but will hold off doing so until hearing from you if think this `spacingSizes` array generation should be happening elsewhere.